packages/artifact/src/internal/upload-http-client.ts

```diff
@@ -55,12 +56,30 @@ export class UploadHttpClient {
    * @returns The response from the Artifact Service if the file container was successfully created
    */
   async createArtifactInFileContainer(
-    artifactName: string
+    artifactName: string,
+    options?: UploadOptions | undefined
   ): Promise<ArtifactResponse> {
     const parameters: CreateArtifactParameters = {
       Type: 'actions_storage',
       Name: artifactName
     }
+
+    // calculate retention period
+    if (options && options.retentionDays) {
+      let retention = options.retentionDays
+      const maxRetentionStr = getRetentionDays()
+      if (maxRetentionStr) {
+        const maxRetention = parseInt(maxRetentionStr)
+        if (!isNaN(maxRetention) && maxRetention < retention) {
+          core.warning(
+            `Retention days is greater than max value allowed by repository setting, reduce retention to ${maxRetention} days`
```

Comment:
Nit: `Retention days is greater than the max value allowed by the repository setting`

Reply:
You don't want to tell the user what the new retention value is set to?  This is the only place user can find out the retention period as of now since we haven't implement any UI change to show artifact expiry.

Reply:
Yes, keep the: `reduce retention to ${maxRetention} days` text at the end, I was mainly nit-picking the missing `the` at the beginning of the warning